src/gengowatcher/browser_session.py

```
@@ -217,6 +244,176 @@ def fetch_browser_session_token_sync(
+        )
+        call_id += 1
+
+        if not isinstance(page_state, dict):
+            raise BrowserSessionError("Browser page state did not return an object")
+
+        job_href = str(page_state.get("jobHref", "") or "").strip()
+        if action == "auto":
+            actions = ["reload", "summary_roundtrip"]
+            if job_href:
+                actions.append("job_roundtrip")
+            action = random.choice(actions)
```

Ruff (0.15.7)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)